### ISSUE TITLE
Searx plugin

### ISSUE THREAD

**Issue by Mikaela** [ProgVal/Limnoria#1238]
This is possibly a wrong place for plugin suggestion, but here it will at least be seen.

As Google has closed their API again, why to not use some more private search engine or meta search engine that is also FOSS and anyone can host their own instance?

I don't know how good this API documentation is https://asciimoo.github.io/searx/dev/search_api.html , but it alows you to get search results in csv/json/rss e.g. https://searx.me/?q=Limnoria&format=json so I imagine it should be possible to get search results out of it in Python (isn't json popular enough).

I have been using Searx.me as my primary search engine for months without issues. I moved there after being suggested it somewhere from DuckDuckGo where if I wanted to get good results out, I had to `!g` to get to Google to find anything.

**Comment by ProgVal** [comment]
That would most likely go to a third-party plugin repo

**Comment by GLolol** [comment]
From what I and @ProgVal discussed earlier it might be beneficial to have a common search API for Limnoria within utils, so plugins [wanting](https://github.com/GLolol/SupyPlugins/blob/5ba2d462738ebf4199a24ba5156b357cbcfbecba/LastFM/plugin.py#L173-L188) [search](https://github.com/ProgVal/Supybot-plugins/blob/e8ecfd0705b23f5283e07954ab6fc89f63d63e15/AutoTrans/plugin.py#L66-L69) [functionality](https://github.com/ddan39/Supybot-IMDb/commit/a2d5eab2c8c8dafb4fba536c1e5b203e21bcbff9) can use whatever search-providing plugin loaded. A possible return result for each search provider could be `search(query) => [{'link': link, 'title': title, 'content': content, …}, …]`

A searx plugin probably wouldn't be that hard to implement, if there's reasonable demand for it :)

**Comment by ProgVal** [comment]
(AutoTrans does not use search)

**Comment by Mikaela** [comment]
I was emailed about https://github.com/horstderheld/limnoria-searx (thanks!), but I think it should be in core (or similar plugin), so I am leaving this open. Feel free to do whatever you think is the best.

**Comment by asterismo** [comment]
hi! i noticed that google search api changed: The Google Web Search API is no longer available. Please migrate to the Google Custom Search API (https://developers.google.com/custom-search/) and the IMDb has changed too: Google search didnt work, returned status 403
any ideas? is there any other updated plugin for this in limnoria? i recently did a git pull and nothing changed
Best regards

**Comment by GLolol** [comment]
@asterismo You have to reinstall & restart the bot after updating the source: `pythonX setup.py install`, where X is the Python version of your preference (2 or 3)

**Comment by asterismo** [comment]
yeah, just upgraded via pip and now it is working. pip3 breaks wikipedia plugin so i only installed limnoria using pip, i was just about to comment

**Comment by GLolol** [comment]
Which wikipedia plugin out of curiosity?

**Comment by ProgVal** [comment]
Please don't use the Google plugin in Limnoria. It's only a temporary solution so people can find other plugins.

**Comment by weedy** [comment]
So what is the google plugin replacement?

**Comment by Mikaela** [comment]
I have been using https://github.com/horstderheld/limnoria-searx

**Comment by ProgVal** [comment]
fyi, we added @jlu5 's DDG plugin in Limnoria a few months ago
